src/main/java/com/google/api/gax/testing/MockStreamObserver.java

```diff
@@ -54,6 +54,7 @@ public void onNext(T message) {
   @Override
   public void onError(Throwable t) {
     errors.add(t);
+    future.set(actualMessages);
```

Comment:
umm using `setException` would throw an exception when getting the response from the future so I was thinking not to throw any exception and use `errors()` to retrieve exceptions later. But now I think maybe throw an exception directly is easier, which makes `errors()` method obsolete.